### PR TITLE
grab visualization improvements

### PR DESCRIPTION
**Pull request by hyperlogic**
* Updated equip logic to use sphere vs sphere tests, to better match visualzation
* Added debug flag for visualizing grab boxes.
* hotspot overlays are now updated as the objects they are attached to move.
* It is easier to near-grab large equippable objects such as the ping-pong gun.
* Optimized EntityPropertyCache to make a single call to Entities.getEntityProperties instead of three.
* Moved grab script options from the "Developer > Hands" menu to the "Developer > Grab Script" menu.

### CODE REVIEW

**Comment by hyperlogic** [comment]
Test plan:
  * Test grab script in demo to make sure it still functions as expected.
  * Test new "Developer > Grab Script > Draw Grab Boxes" menu option.  This option should draw a blue sphere around your hand when you squeeze the trigger.  When the blue sphere overlaps with the blue box of an entity, you will near grab it.  When the green sphere overlaps the green sphere on the object you will equip it instead.
  * It should be possible to near-grab the ping-pong gun by grabbing the stock or the barrel, were previously this was difficult.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 7bd553c09cea7784828354085901adf788f37a0c available at: https://gist.github.com/bb62163180823dd59afb2fe9df326a86<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8127.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8127.dmg

File: scripts/system/controllers/handControllerGrab.js [hunk 1]
```diff
@@ -872,43 +869,147 @@ function MyController(hand) {
...
+                color: HAND_EQUIP_SPHERE_COLOR,
+                alpha: HAND_EQUIP_SPHERE_ALPHA,
+                solid: true,
+                visible: true,
+                ignoreRayIntersection: true,
+                drawInFront: false
+            });
+
+            this.hotspotOverlays.push({
+                entityID: undefined,
```
Comment (howard-stearns) [inline scripts/system/controllers/handControllerGrab.js]: This line is effectively just a code comment?

File: scripts/system/libraries/Xform.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
+//
```
Comment (howard-stearns) [inline scripts/system/libraries/Xform.js]: Nice! But where is this file used?

File: scripts/system/controllers/handControllerGrab.js [hunk 3]
```diff
@@ -872,43 +869,147 @@ function MyController(hand) {
...
-        for (i = 0; i < l; i++) {
-            Overlays.deleteOverlay(this.equipHotspotOverlays[i]);
-        }
-        this.equipHotspotOverlays = [];
+    this.destroyHotspots = function() {
+        this.hotspotOverlays.forEach(function (overlayInfo) {
+            Overlays.deleteOverlay(overlayInfo.overlay);
+        });
+        this.hotspotOverlays = [];
+    };
```
Comment (howard-stearns) [inline scripts/system/controllers/handControllerGrab.js]: Should we connect Script.scriptEnding to destroyHotspots? (And possibly other similar methods, if there are any.)

**Comment by howard-stearns** [comment]
I will QA this in the morning.

**Comment by RyanDowne** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;Y3W2rag1&#x2F;54-update-green-equip-ball-as-object-moves"> Update green equip ball as object moves.</a>

**Comment by RyanDowne** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;R0ivOgPz&#x2F;56-cannot-near-grab-even-large-equipable-objects"> Cannot near grab even large equipable objects.</a>

File: scripts/system/controllers/handControllerGrab.js (hunk 1 above)
Comment (hyperlogic) [inline scripts/system/controllers/handControllerGrab.js]: No, DRAW_HAND_SPHERES is set to true when you enable the "Developer > Show Grab Boxes" menu option

File: scripts/system/libraries/Xform.js (hunk 2 above)
Comment (hyperlogic) [inline scripts/system/libraries/Xform.js]: Ah, I was using it, but I'm not at the moment....  I guess I should have removed it.

File: scripts/system/controllers/handControllerGrab.js (hunk 1 above)
Comment (howard-stearns) [inline scripts/system/controllers/handControllerGrab.js]: (From discussion: the entityID ends up being in the enumerated properties, and the intent was for that to be the case.)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 41fcfb45caf4592c0020d89a7149967b56932edd available at: https://gist.github.com/bb62163180823dd59afb2fe9df326a86<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8127.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8127.dmg

**Comment by howard-stearns** [comment]
Works well. I still don't think that drop without shaking is working, but that's not changed in this PR.

**Comment by howard-stearns** [comment]
I'll merge with the unused file, but we should use it or lose it.

**Pull request merged**